src/codegen/generators/pou_generator.rs

```diff
@@ -61,6 +61,74 @@ pub fn generate_implementation_stubs<'ink>(
     Ok(llvm_index)
 }
 
+pub fn generate_global_constants_for_pou_members<'ink>(
+    module: &Module<'ink>,
+    llvm: Llvm<'ink>,
+    index: &Index,
+    annotations: &AstAnnotations,
+    llvm_index: &LlvmTypedIndex<'ink>,
+) -> Result<LlvmTypedIndex<'ink>, Diagnostic> {
+    let mut local_llvm_index = LlvmTypedIndex::default();
+    for (_, implementation) in index.get_implementations() {
+        let type_name = implementation.get_type_name();
+        let pou_members = index.get_container_members(type_name);
+        let variables = pou_members
+            .iter()
+            .filter(|it| it.is_local() || it.is_temp())
+            .filter(|it| {
+                let var_type = index
+                    .get_effective_type_by_name(it.get_type_name())
+                    .get_type_information();
+                var_type.is_struct() || var_type.is_array() || var_type.is_string()
+            });
+        let exp_gen =
+            ExpressionCodeGenerator::new_context_free(&llvm, index, annotations, llvm_index);
+        for variable in variables {
+            let name = format!("{}__init", variable.get_qualified_name());
+            let right_stmt = match variable.initial_value {
+                Some(..) => Some(
+                    index
+                        .get_const_expressions()
+                        .maybe_get_constant_statement(&variable.initial_value)
+                        .ok_or_else(|| {
+                            Diagnostic::cannot_generate_initializer(
+                                variable.get_qualified_name(),
+                                variable.source_location.clone(),
+                            )
+                        })?,
+                ),
+                None => None,
+            };
+
+            if let Some(stmt) = right_stmt {
+                if llvm_index.find_global_value(&name).is_none() {
+                    //Get either a global initial value for the constant (For arrays) and copy it,
+                    let value = if let Some(value) =
+                        llvm_index.find_associated_initial_value(variable.get_qualified_name())
+                    {
+                        value
+                    } else {
+                        exp_gen.generate_expression(stmt)?
+                    };
+                    let variable_type = llvm_index.get_associated_type(variable.get_type_name())?;
+                    //First try to find a global with that name
+                    let global_value = module.get_global(&name).unwrap_or_else(|| {
```

Comment:
didnt we establish in line 104 that there is no global with that name? Why do you expect one in the module?

Reply:
True, I think this was leftover from a previous try. Removing it didn't affect the tests.